closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -162,19 +165,31 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_DigestComputeAnd
     unsigned char* digestNative;
 
     if (message != NULL) {
-       messageNative = (*env)->GetPrimitiveArrayCritical(env, message, &isCopy);
-       messageNative = messageNative + messageOffset;
-       if (1 != EVP_DigestUpdate(context->ctx, messageNative, messageLen)) handleErrors();
-           (*env)->ReleasePrimitiveArrayCritical(env, message, NULL, 0);
+        messageNative = (*env)->GetPrimitiveArrayCritical(env, message, &isCopy);
+
+        if (messageNative == NULL)
+            return -1;
+
+        messageNative = messageNative + messageOffset;
```

Comment:
This line modify an original messageNative so it can not be used in ReleasePrimitiveArrayCritical()

Reply:
Oh.. Yes! I will address this.